InfiniteRecharge/src/main/java/frc/robot/subsystem/scoring/shooter/ShooterSubsystem.java

```diff
@@ -127,4 +132,23 @@ public void rotateToDeg(double targetPoint) {
         targetChange = 0;
     }
 
-}
\ No newline at end of file
+    public void rotateByDeg(double degrees) {
+        rotateToDeg(getTargetTurretDegGivenOffset(degrees));
+    }
+    
+    public double getTurretDeg() {
+        double encoderDeg =  azimuthMotor.getSelectedSensorPosition() * (360.0 / config.shooter.azimuth.ticksPerRevolution);
+        double turretDeg = encoderDeg * config.shooter.gearRatio;
+        return turretDeg;
+    }
+
+    public double getTargetTurretDegGivenOffset(double offset) {
+        return getTurretDeg() - offset;
+    }
+
+    public void rotateTurretGivenLLOffset(double offset) {
+        double avgOffset = filter.calculate(getTargetTurretDegGivenOffset(offset));
+        rotateByDeg(avgOffset);

```

Comment:
so getTargetTurretDegGivenOffset returns a turret angle, not an offset, to command so when filtered it will return the same thing, an turret angle instead of an offset, so avgOffset should be renamed accordingly. Also, since it is not an offset and rather an angle, you should call rotateToDeg()